sequencer/src/block.rs

```diff
@@ -454,3 +492,209 @@ mod reference {
         );
     }
 }
+
+#[cfg(test)]
+mod test_headers {
+    use super::*;
+    use async_compatibility_layer::logging::{setup_backtrace, setup_logging};
+
+    /// Just the part of the header we care about for these tests.
+    #[derive(Debug, Default)]
+    struct HeaderInfo {
+        timestamp: u64,
+        l1_head: u64,
+        l1_finalized: Option<L1BlockInfo>,
+    }
+
+    #[derive(Debug, Default)]
+    #[must_use]
+    struct TestCase {
+        parent: HeaderInfo,
```

Comment:
Not super important but I think the test cases would be shorter easier to read (less nesting and `..Default::default()`) if the struct here was flattened instead of nested.